STM32WB IWDG Support

robbawebba opened the pull request:
The STM32WB SoC supports an independent watchdog that is compatible with the existing stm32 watchdog driver and watchdog API. This pull request adds watchdog support for this SoC for use with the watchdog drivers as well as enables watchdog support for the nucleo_wb55rg board.

This has been tested with the nucleo_wb55rg and the sample located at `samples/drivers/watchdog`.

robbawebba committed:
soc: stm32wb: Add support for STM32 IWDG

The driver for STM32's independent watchdog already exists and is
compatible with the stm32wb SoC. Enable the independent watchdog
for the stm32wb series for use with this driver.

Signed-off-by: Robert Weber <robertweber95@gmail.com>

robbawebba committed:
boards: nucleo_wb55rg: Enable IWDG

Enable the independent watchdog on the nucleo_wb55rg

Signed-off-by: Robert Weber <robertweber95@gmail.com>

robbawebba commented:
I noticed that the file `boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst` does not contain an underscore in the board name (i.e. `nucleo_wb55rg.rst`). It doesn't seem like too big of an issue, but I wanted to double check with you in case either of you had a preference. Is this alright, or is this something that you would like changed?

erwango commented on soc/arm/st_stm32/stm32wb/dts_fixup.h:
```diff
@@ -155,4 +155,6 @@
 #define DT_ADC_1_CLOCK_BITS		DT_ST_STM32_ADC_50040000_CLOCK_BITS_0
 #define DT_ADC_1_CLOCK_BUS		DT_ST_STM32_ADC_50040000_CLOCK_BUS_0
 
+#define DT_WDT_0_NAME         DT_INST_0_ST_STM32_WATCHDOG_LABEL
```
Please keep indentation

robbawebba replied:
Addressed, thanks!

erwango commented on soc/arm/st_stm32/stm32wb/dts_fixup.h:
```diff
@@ -155,6 +155,6 @@
 #define DT_ADC_1_CLOCK_BUS		DT_ST_STM32_ADC_50040000_CLOCK_BUS_0
 
-#define DT_WDT_0_NAME         DT_INST_0_ST_STM32_WATCHDOG_LABEL
+#define DT_WDT_0_NAME			DT_INST_0_ST_STM32_WATCHDOG_LABEL
```
Well, move this fix in previous commit and we're good

robbawebba replied:
Addressed, thanks for catching that!

robbawebba replied:
Addressed, thanks!

erwango reviewed (changes requested):
Minor remark

dbkinder reviewed (changes requested):
It would indeed be good to rename the document to ``nucleo_wb55rg.rst``, otherwise, LGTM.

robbawebba committed:
boards: nucleo_wb55rg: Rename documentation file to match board name

Rename doc/nucleowb55rg.rst to doc/nucleo_wb55rg.rst so the file name
matches the name of the board it is documenting.

Signed-off-by: Robert Weber <robertweber95@gmail.com>

robbawebba commented:
@dbkinder and @erwango Thanks for the feedback! I made the requested changes. Could you please check that the changes meet your expectations? Thanks in advance!

dbkinder reviewed (approved):
doc changes LGTM, thanks.
